src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -20,13 +20,15 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 
 	private readonly collection: Collection;
 	private readonly _query: object | undefined;
+	private readonly _projection: object | undefined;
 	private _cursor: Cursor | undefined;
 	private _hasMoreChildren: boolean = true;
 	private _batchSize: number = DefaultBatchSize;
 
-	constructor(collection: Collection, query?: string) {
+	constructor(collection: Collection, query?: string[]) {
 		this.collection = collection;
-		this._query = query ? JSON.parse(query) : undefined;
+		this._query = query && query.length > 0 ? JSON.parse(query[0]) : undefined;
+		this._projection = query && query.length >= 2 ? JSON.parse(query[1]) : undefined;
```

Comment:
I wanted to convey a 'length of at least 2' succinctly. Would this be clearer or `query.length > 1` be?